# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gcc/sparc-no-app-regs.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Using the application preserved registers introduces quite some packaging
nightmare, since no .so can be assembled using those ...


  - Rene Rebe <rene@exactcode.de>

--- ./gcc/config/sparc/linux64.h.orig	2005-01-19 01:07:56.434491448 +0100
+++ ./gcc/config/sparc/linux64.h	2005-01-19 01:30:08.454993552 +0100
@@ -48,7 +48,7 @@
 #undef TARGET_DEFAULT
 #define TARGET_DEFAULT \
   (MASK_V9 + MASK_PTR64 + MASK_64BIT /* + MASK_HARD_QUAD */ \
-   + MASK_STACK_BIAS + MASK_APP_REGS + MASK_FPU + MASK_LONG_DOUBLE_128)
+   + MASK_STACK_BIAS + MASK_FPU + MASK_LONG_DOUBLE_128)
 #endif
 
 #undef ASM_CPU_DEFAULT_SPEC


Commented out for now ...
C  --- ./gcc/config/sparc/sparc.h.orig	2005-01-19 01:25:53.808705672 +0100
C  +++ ./gcc/config/sparc/sparc.h	2005-01-19 01:29:17.431750264 +0100
C  @@ -617,7 +617,7 @@
 /* MASK_APP_REGS must always be the default because that's what
    FIXED_REGISTERS is set to and -ffixed- is processed before
    CONDITIONAL_REGISTER_USAGE is called (where we process -mno-app-regs).  */
-#define TARGET_DEFAULT (MASK_APP_REGS + MASK_FPU)
+#define TARGET_DEFAULT (MASK_FPU)
 
 /* This is meant to be redefined in target specific files.  */
 #define SUBTARGET_SWITCHES
